edge/pkg/devicetwin/dtmanager/twin_test.go

```diff
@@ -0,0 +1,1604 @@
+			errwant:  nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if _, err := dealTwinSync(test.context, test.resource, test.msg); !reflect.DeepEqual(err, test.errwant) {
+				t.Errorf("DTManager.dealTwinSync() error = %v, Want = %v", err, test.errwant)
+			}
+		})
+	}
+}
+
+func TestDealTwinGet(t *testing.T) {
+	content, _ := json.Marshal(keyTwinUpdateFunc())
+	context := contextFunc()
+
+	tests := []struct {
+		name     string
+		context  *dtcontext.DTContext
+		resource string
+		msg      interface{}
+		errwant  error
+	}{
+		{
+			name:     "TestDealTwinGet(): Case 1: msg not Message type",
+			context:  &dtcontext.DTContext{},
+			resource: "",
+			msg:      msgFunc1(dttype.BaseMessage{EventID: "Event1"}),
+			errwant:  errors.New("msg not Message type"),
+		},
+		{
+			name:     "TestDealTwinGet(): Case 2: invalid message content",
+			context:  &dtcontext.DTContext{},
+			resource: "",
+			msg:      msgFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errwant:  errors.New("invalid message content"),
+		},
+		{
+			name:     "TestDealTwinGet(): Case 3: Success case",
+			context:  &context,
+			resource: "DeviceB",
+			msg:      msgFunc(content),
+			errwant:  nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if _, err := dealTwinGet(test.context, test.resource, test.msg); !reflect.DeepEqual(err, test.errwant) {
+				t.Errorf("DTManager.dealTwinGet() error = %v, Want = %v", err, test.errwant)
+			}
+		})
+	}
+}
+
+func TestDealTwinUpdate(t *testing.T) {
+	content, _ := json.Marshal(dttype.DeviceTwinUpdate{BaseMessage: dttype.BaseMessage{EventID: "Event1"}})
+	context := contextFunc()
+
+	tests := []struct {
+		name     string
+		context  *dtcontext.DTContext
+		resource string
+		msg      interface{}
+		errwant  error
+	}{
+		{
+			name:     "TestDealTwinUpdate(): Case 1: msg not Message type",
+			context:  &dtcontext.DTContext{},
+			resource: "",
+			msg:      msgFunc1(dttype.BaseMessage{EventID: "Event1"}),
+			errwant:  errors.New("msg not Message type"),
+		},
+		{
+			name:     "TestDealTwinUpdate(): Case 2: invalid message content",
+			context:  &dtcontext.DTContext{},
+			resource: "",
+			msg:      msgFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errwant:  errors.New("invalid message content"),
+		},
+		{
+			name:     "TestDealTwinUpdate(): Case 3: Success case",
+			context:  &context,
+			resource: "DeviceB",
+			msg:      msgFunc(content),
+			errwant:  nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if _, err := dealTwinUpdate(test.context, test.resource, test.msg); !reflect.DeepEqual(err, test.errwant) {
+				t.Errorf("DTManager.dealTwinUpdate() error = %v, Want = %v", err, test.errwant)
+			}
+		})
+	}
+}
+
+func TestUpdated(t *testing.T) {
+	content, _ := json.Marshal(dttype.DeviceTwinUpdate{BaseMessage: dttype.BaseMessage{EventID: "Event1"}})
+	content1, _ := json.Marshal(keyTwinUpdateFunc())
```

Comment:
lets rename to something logical and more descriptive instead of 1,2...
Applicable to all similar cases below